src/phoenix/server/app.py

```diff
@@ -192,6 +193,16 @@ def create_app(
                 ),
             ]
         )
+        + (
+            []
+            if traces is None
+            else [
+                Route(
+                    "/v1/traces",
+                    type("TraceEndpoint", (TraceHandler,), {"queue": traces}),
+                ),
+            ]
+        )
```

Comment:
I'm a bit confused by this here - are these endpoints not mounted if traces is none? That doesn't make sense to me...

Reply:
If it's `None` then we don't have a queue to put the traces in, because the queue resides with the traces object. In that case it would just return 500, so it's the same as not mounting.

Reply:
Okay I think this diverged from my understanding - didn't realize we stopped consuming if we mounted datasets. We can revisit later howerver.

Reply:
`Traces` is always [initialized](https://github.com/Arize-ai/phoenix/blob/main/src/phoenix/session/session.py#L117) so it's never None. It's just that the type signature says [Optional](https://github.com/Arize-ai/phoenix/blob/9ddbd9caa4a66ae47c5447aa4dafa46cc9c5cfa5/src/phoenix/server/app.py#L163).